pkg/controllers/user/serviceaccount/serviceaccount.go

```diff
@@ -0,0 +1,343 @@
+				}
+			}
+
+			if policy == nil {
+				// get PSPT
+				// if no PSPT then get default
+				// if no default then exit
+				namespaces, err := m.namespaces.List("", labels.Everything())
+
+				if err != nil {
+					return fmt.Errorf("error getting namespaces: %v", err)
+				}
+
+				var namespace *v1.Namespace
+
+				for _, candidate := range namespaces {
+					if candidate.Name == namespaceName {
+						namespace = candidate
+					}
+				}
+
+				if namespace == nil {
+					return fmt.Errorf("no namespace exists with namespace of service account: %v", namespaceName)
+				}
+
+				annotation := namespace.Annotations[projectIdAnnotation]
+
+				if annotation == "" {
+					// no project is associated with namespace so don't do anything
+					logrus.Infof("no project is associated with namespace %v", namespaceName)
+					return nil
+				}
+
+				// TODO what if we can't parse this?
+				split := strings.Split(annotation, ":")
+				clusterName, projectId := split[0], split[1]
+
+				projects, err := m.projects.List("", labels.Everything())
+
+				if err != nil {
+					return fmt.Errorf("error getting projects: %v", err)
+				}
+
+				var project *v3.Project
+
+				for _, candidate := range projects {
+					if candidate.Name == projectId {
+						project = candidate
+					}
+				}
+
+				if project == nil {
+					return fmt.Errorf("no project with project id %v was found", projectId)
+				}
+
+				podSecurityPolicyTemplateId := project.Annotations[podSecurityPolicyTemplateLabel]
+
+				if podSecurityPolicyTemplateId == "" {
+					// check cluster
+					clusters, err := m.clusters.List("", labels.Everything())
+
+					if err != nil {
+						return fmt.Errorf("error getting clusters: %v", err)
+					}
+
+					var cluster *v3.Cluster
+
+					for _, candidate := range clusters {
+						if candidate.Name == clusterName {
+							cluster = candidate
+						}
+					}
+
+					if cluster == nil {
+						return fmt.Errorf("could not find cluster %v", clusterName)
+					}
+
+					podSecurityPolicyTemplateId = cluster.Annotations[podSecurityPolicyTemplateLabel]
+
+					if podSecurityPolicyTemplateId == "" {
+						logrus.Infof("START CLUSTER")
+						logJson(cluster)
+						logrus.Infof("END CLUSTER")
+						logrus.Infof("START PROJECT")
+						logJson(project)
+						logrus.Infof("END PROJECT")
+
+						logrus.Infof("No PSPTs found for project %v and cluster %v", projectId, clusterName)
+						return nil
+					}
+				}
+
+				templates, err := m.templates.List("", labels.Everything())
+
+				if err != nil {
+					return fmt.Errorf("error getting pod security policy templates: %v", err)
+				}
+
+				var template *v3.PodSecurityPolicyTemplate
+
+				for _, candidate := range templates {
+					if candidate.Name == podSecurityPolicyTemplateId {
+						template = candidate
+					}
+				}
+
+				if template == nil {
+					return fmt.Errorf("no template exists with id %v", podSecurityPolicyTemplateId)
+				}
+
+				template.Kind = "PodSecurityPolicy"
```

Comment:
very important: you cannot modify objects you got directly from the cache/lister. It changes that reference for all controllers using this cache/lister.
you have to do template.DeepCopy() and modify the returned object

Reply:
But also, dont reuse template.typemeta or template.objectmeta from the PSPT to create the PSP. I think that will bite us eventually, just create new ones. you can do it in line when creating the struct like:
policy := PSP{
  ObjectMeta: v1meta.ObjectMeta{...}
}
```